// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package zen_data_svr;

//  
service ZendataHandle {
  rpc PushKlineData (PushKlineReq) returns (PushKlineRsp) {}
  rpc AskKlineData (AskKlineReq) returns (AskKlineRsp) {}
}

//  
message PushKlineReq {
  int32 ts = 1;
  string code = 2;
  string sec_type = 11;
  string period = 3;
  double open = 4;
  double high = 5;
  double low = 6;
  double close = 7;
  int64 vol = 8;
  double amount = 9;
  int32 is_end = 10;
}

//  
message PushKlineRsp {
  int32 status = 1;
  string message = 2;
}

//  
message AskKlineReq {
  string code = 1;
  string period = 2;
  string sec_type = 3;
  int32  from_ts = 4;
  int32  to_ts = 5;
}

message Kline{
  int32  ts = 1;
  double open = 2;
  double high = 3;
  double low = 4;
  double close = 5;
  int64  vol = 6;
  double amount = 7;
  int32  is_end = 8;
}

message AskKlineRsp {
  int32  status = 1;
  string message = 2;
  string code = 3;
  string period = 4;
  string sec_type = 5;
  repeated Kline klines = 6;
}

